Fix polymer tutorial

florian-f opened the pull request:
https://github.com/gwtproject/gwt-site/issues/114

Added a working pom.xml and some missing imports/ other small stuff.

It would be nice if someone could verify that this works for them.

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/gwtproject/gwt-site/116)
<!-- Reviewable:end -->

tbroyer commented:
I don't understand the part about the maven-war-plugin and maven-resources-plugin.

florian-f commented:
> I don't understand the part about the maven-war-plugin and maven-resources-plugin.

the resources (most importantly `src/main/webapp/TodoList.html`) aren't copied to the output directory without them. 

I suspect that this is another bug in webAppCreator. 

To clarify: If you 

* create a project with `webAppCreator -out Test1 -templates maven,sample org.test.Test` 
* and fix the pom.xml by adding `<type>pom</type>
* and then run it with `mvn gwt:run`

you will get a 404 when you go to `http://127.0.0.1:8888/Test.html`. This happens because the contents of `src/main/webapp/` aren't copied into `target/Test-1.0-SNAPSHOT/`.

This can be fixed by adding these plugins. I got this from the [pom.xml in the gwt-polymer-elements demo](https://github.com/vaadin/gwt-polymer-elements/blob/master/demo/pom.xml).

florian-f commented:
To be honest, it would be easier to set up the project without webAppCreator.

tbroyer commented:
You have to run `mvn package` or `mvn war:exploded` before `mvn gwt:run`; this is not new, and independent from whether you use webAppCreator or not.

florian-f commented:
You are absolutely right of course. I removed the plugins (and the `pom.xml`) from `create.md` and added a line telling people to run `mvn war:exploded` before running `mvn gwt:run`.

manolo commented on src/main/markdown/doc/latest/polymer-tutorial/create.md:
```diff
@@ -33,12 +33,14 @@ For the **TodoList** project, we'll need to run `webAppCreator` with the followi
       _**Tip**: If you include the GWT SDK folder in your PATH environment variable, you won't have to specify the full path._
 
+      You may have to modify the `pom.xml` before you can run the application. Add `<type>pom</type>` to the gwt dependency, otherwise you will encounter an error. 
 
```
This is already documented and explained in : http://www.gwtproject.org/doc/latest/RefCommandLineTools.html#webAppCreator , can you add a link to it?

florian-f replied:
done

manolo commented on src/main/markdown/doc/latest/polymer-tutorial/create.md:
```diff
@@ -33,12 +33,14 @@ For the **TodoList** project, we'll need to run `webAppCreator` with the followi
      To check that the project was created correctly start the new app in SuperDevMode.
 
         $ cd TodoListApp
+        $ mvn war:exploded
```
It could be better to do all in one line, so the user never forgets to copy resources:
    
     $ mvn war:exploded gwt:run

florian-f replied:
good point, although the user only needs to run `mvn war:exploded` _once_ unless he runs `mvn clean` at some point, so I think it's safe. Alternatively, providing them with a `pom.xml` that just includes the `maven-war-plugin` (which is what `mvn war:exploded` calls) like you do [in the gwt-polymer-elements demo](https://github.com/vaadin/gwt-polymer-elements/blob/master/demo/pom.xml) would be a way to make absolutely sure the resources get copied.

manolo replied:
Including maven-war-plugin could be a good idea but for the webAppCreator instead. We don't want the user to modify so much the generated pom in this tutorial.
So, to me is safer to do `mvn war:exploded gwt:run` all in one line to avoid problems if he runs clean or he modifies the static stuff in webapp folder.

manolo commented:
LGTM, a couple of nits though
